bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -1041,11 +1041,14 @@ func (r *KubeadmConfigReconciler) ensureBootstrapSecretOwnersRef(ctx context.Con
 	if err != nil {
 		return errors.Wrapf(err, "failed to add KubeadmConfig %s as ownerReference to bootstrap Secret %s", scope.ConfigOwner.GetName(), secret.GetName())
 	}
+	if controller := metav1.GetControllerOf(secret); controller != nil {
```

Comment:
Not sure if this one is necessary.

I think in only looked like this in the reported error as the new ownerRef was the wrong one.

Should be safer to have it anyway though

Reply:
Yeah - it's not necessary to fix this, but it might be a good guard / assertion of the proper chain of ownership.

Reply:
Sounds fine!!